engine/src/test/java/io/zeebe/engine/processing/streamprocessor/StreamProcessorTest.java

```diff
@@ -673,6 +673,55 @@ public void onResumed() {
         .isEqualTo(positionProcessedAfterResume);
   }
 
+  @Test
+  public void shouldNotOverwriteLastWrittenPositionIfNoFollowUpEvent()
+      throws ExecutionException, InterruptedException {
+    // given
+    final var onProcessedListener = new AwaitableProcessedListener();
+    streamProcessorRule.startTypedStreamProcessor(
+        (processors, state) ->
+            processors
+                .onEvent(
+                    ValueType.WORKFLOW_INSTANCE,
+                    WorkflowInstanceIntent.ELEMENT_ACTIVATING,
+                    new TypedRecordProcessor<>() {
+                      @Override
+                      public void processRecord(
+                          final long position,
+                          final TypedRecord<UnifiedRecordValue> record,
+                          final TypedResponseWriter responseWriter,
+                          final TypedStreamWriter streamWriter,
+                          final Consumer<SideEffectProducer> sideEffect) {
+                        streamWriter.appendFollowUpEvent(
+                            record.getKey(),
+                            WorkflowInstanceIntent.ELEMENT_ACTIVATED,
+                            record.getValue());
+                      }
+                    })
+                .onEvent(
+                    ValueType.WORKFLOW_INSTANCE,
+                    WorkflowInstanceIntent.ELEMENT_ACTIVATED,
+                    mock(TypedRecordProcessor.class))
+                .onEvent(
+                    ValueType.WORKFLOW_INSTANCE,
+                    WorkflowInstanceIntent.CANCEL,
+                    mock(TypedRecordProcessor.class)),
+        onProcessedListener.expect(2));
+
+    // when
+    final long position =
+        streamProcessorRule.writeWorkflowInstanceEvent(WorkflowInstanceIntent.ELEMENT_ACTIVATING);
+    assertThat(onProcessedListener.await()).isTrue();
+    onProcessedListener.expect(1);
+    streamProcessorRule.writeWorkflowInstanceEvent(WorkflowInstanceIntent.CANCEL);
+    assertThat(onProcessedListener.await()).isTrue();
```

Comment:
(optionally) We can separate the two cases to improve the readability. 

```suggestion
    final long position =
        streamProcessorRule.writeWorkflowInstanceEvent(WorkflowInstanceIntent.ELEMENT_ACTIVATING);
    assertThat(onProcessedListener.await()).isTrue();
    
    onProcessedListener.expect(1);
    streamProcessorRule.writeWorkflowInstanceEvent(WorkflowInstanceIntent.CANCEL);
    assertThat(onProcessedListener.await()).isTrue();
```